pkg/opensearch/lucene_handler.go

```diff
@@ -57,26 +59,31 @@ func (h *luceneHandler) processQuery(q *Query) error {
 	b := h.ms.Search(interval)
 	b.Size(0)
 
-
 	filters := b.Query().Bool().Filter()
 	defaultTimeField := h.client.GetConfiguredFields().TimeField
 
 	if q.luceneQueryType == luceneQueryTypeTraces {
 		traceId := getTraceId(q.RawQuery)
-		if traceId != "" {
-			b.Size(1000)
-			b.SetTraceSpansFilters(toMs, fromMs, traceId)
-		} else {
-			b.SetTraceListFilters(toMs, fromMs, q.RawQuery)
+		switch q.serviceMapInfo.Type {
+		case Prefetch:
+			b.Size(0)
```

Comment:
kinda silly question on my part, why is this zero? Like we want to get some amount of results no?

Reply:
The size parameter is used to determine how many `hits` are returned in the response. Meaning, documents that match the whatever filters, range etc. the query sends. Since we only need aggregations (`buckets` field) in this response, we don't need or use the `hits` field.